### ISSUE TITLE
Why we need to call util.attach_to_log for warning and error messages?

### ISSUE THREAD

**Issue by wkentaro** [mikedh/trimesh#396]
Currently, most of the warning and error messages are hidden unless using `util.attach_to_log`, for example:
https://github.com/mikedh/trimesh/blob/08507c9b1b69c61f26a56a74d13187f3d2602293/trimesh/exchange/wavefront.py#L177-L178

Normally, I think this behavior is undesired because we sometimes make mistakes (e.g., forget installing Pillow), and that causes issues that are hard to fix (e.g., texture is not loaded and displayed). A solution is we call `util.attach_to_log` in every script, but we don't want to do that (at least I don't).

IMO, I think we should show warnings and error messages as default (without calling `util.attach_to_log` like some of the warnings like:

https://github.com/mikedh/trimesh/blob/323a6f0e503b05cefbe11fbb8eedb5705fb7d01f/trimesh/points.py#L362-L366

What do you think of this?

**Comment by wkentaro** [comment]
Probably, one option is attaching to the `logger` of `trimesh` at the level of `warning` as default. By doing this, it shows `warning`, `error`, and `fatal` messages without calling `attach_to_log`.

**Comment by mikedh** [comment]
Oh yeah, it seems like any `warnings.warn` should be printed. I wonder why the warnings from inside trimesh are hidden, when I call `warnings.warn` from an IPython terminal they show up:

```
In [1]: import warnings

In [2]: warnings.warn('hi')
/home/mikedh/miniconda/bin/ipython:1: UserWarning: hi
  #!/home/mikedh/miniconda/bin/python

In [3]: import trimesh

In [4]: warnings.warn('hi')
/home/mikedh/miniconda/bin/ipython:1: UserWarning: hi
  #!/home/mikedh/miniconda/bin/python

In [5]: trimesh.util.attach_to_log()

In [6]: warnings.warn('hi')
WARNING        warnings.py:99    /home/mikedh/miniconda/bin/ipython:1: UserWarning: hi
  #!/home/mikedh/miniconda/bin/python

```

**Comment by wkentaro** [comment]
With below change, the `warnings` is shown.
```
diff --git a/trimesh/constants.py b/trimesh/constants.py
index 872c6ee..650040b 100644
--- a/trimesh/constants.py
+++ b/trimesh/constants.py
@@ -126,7 +126,7 @@ res_path = ResolutionPath()
 
 # logging
 log = _getLogger('trimesh')
-log.addHandler(_NullHandler())
+# log.addHandler(_NullHandler())
 
 
 def log_time(method):
diff --git a/trimesh/util.py b/trimesh/util.py
index b1dc318..ef6a89f 100644
--- a/trimesh/util.py
+++ b/trimesh/util.py
@@ -36,7 +36,7 @@ else:
 
 # create a default logger
 log = logging.getLogger('trimesh')
-log.addHandler(logging.NullHandler())
+# log.addHandler(logging.NullHandler())
 
 # include constants here so we don't have to import
 # a floating point threshold for 0.0
```

```
#!/usr/bin/env python

import numpy as np
import trimesh

points = np.random.uniform(-0.3, 0.3, (128, 3))
colors = np.random.uniform(0, 1, (128, 3))
geom = trimesh.PointCloud(vertices=points, color=colors)
```

```
$ PYTHONWARNINGS=default::DeprecationWarning python spam.py
No FCL -- collision checking will not work
/home/wkentaro/trimesh/.anaconda3/lib/python3.6/site-packages/future/standard_library/__init__.py:65: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
/home/wkentaro/Documents/trimesh/trimesh/points.py:366: DeprecationWarning: 'PointCloud(color=__)' is deprecated and will be removed in versions released after 6/1/2019. Use 'PointCloud(colors=__)' instead
  warnings.warn(msg, DeprecationWarning)

```

**Comment by mikedh** [comment]
Fixed by #402. Thanks!
